frontend/src/views/Game.tsx

```diff
@@ -104,8 +133,28 @@ function GamePage() {
 
   // Callback when user runs code against custom test case
   const runSolution = (input: string) => {
-    const tempSubmission = { status: 'SUCCESS', output: input };
-    setSubmission(tempSubmission);
+    const request = {
+      initiator: currentUser!,
+      code: input,
+      language: currentLanguage,
+    };
+
+    submitSolution(roomId, request)
+      .then((res) => {
+        setSubmission(res);
+      })
+      .catch((err) => setError(err));
+  };
+
+  const displayPlayerLeaderboard = () => {
+    return players.map((player, index) => (
+      <LeaderboardCard
+        player={player}
+        isCurrentPlayer={player.user.userId === currentUser?.userId}
+        place={index + 1}
+        color="blue" // TODO: merge with Chris's color PR
+      />
```

Comment:
The leaderboard doesn't seem to update correctly? It shifts briefly when the user first submits, then shifts back. See the [attached screencast](https://www.loom.com/share/6310b20e52844c6b9f1fe23c4b683de5).

Reply:
Nice catch, at first I thought it had to do with the circular thing mentioned above but I got rid of it and still have some weird subscription behavior going on in the console (which I assume has something to do with it shifting back): 

<img width="200" alt="Screenshot" src="https://user-images.githubusercontent.com/24768574/104886615-e7560f80-591e-11eb-9643-f426a8c64e92.png">

I'll try to figure it out tmrw but if I can't maybe we can do some pair programming haha

Reply:
This sounds good -- just let me know, I'm definitely open to pair programming.

I do remember I got some sort of circular issue as well at some point, but I didn't mention it because I couldn't replicate it. We can try to figure this out sometime today if you want.

Reply:
Was able to fix it - I got rid of the useCallback for setStateFromGame and simply moved the setCurrentPlayer logic to a separate useEffect block - this seemed to get rid of the constant resubscribing

Reply:
I'll resolve this as I think the leaderboard bug should be fixed, but feel free to verify on your end as well

Reply:
The bug is fixed, but there's another note -- this might depend on the game type or settings we choose to implement, but if Player 1 submits correctly, then Player 2 submits correctly, then Player 1 submits correctly _again_, the leaderboard puts Player 1 **behind** Player 2. I'm not sure if this is intentional or not because Player 1 already got the best solution, but I wanted to note this down and see if we should change this @alankbi.

Reply:
Yeah so the current implementation is by the latest submission only - so if someone gets 100% but decide to submit again then their latest submission would count (and penalize them for time of submission). If we want to change this I can think of two alternatives (although there might be better ones out there): 1. change implementation to rank people by their best solutions, 2. prevent additional submissions once people finish solving the problem. Thoughts?

Reply:
Those are good alternatives -- I think I might lean towards #1? Since I can see preventing additional submissions in some formats but I think it'd be good to stay consistent across most formats that the best solution is the one that is preserved.

Reply:
Sounds good - implemented